src/utils/test.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::{BTreeMap, HashSet};
```

Comment:
I used these helper functions to make testing the above easier. I think they have value for writing tests for other portions of our codebase though so I'm leaving them in.